### DISCUSSION TITLE
Firebase_Messaging getToken error.

### DISCUSSION THREAD

**Discussion by codeTerrain** [firebase/flutterfire#13578] [Q&A]
I am trying to call the Firebase_messaging getToken() function to get the token for my device, but I am picking up this error. I am using Flutter 3. Anyone to help please?

E/flutter (  696): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: [firebase_messaging/unknown] java.io.IOException: java.util.concurrent.ExecutionException: java.io.IOException: INVALID_PARAMETERS
E/flutter (  696): #0      StandardMethodCodec.decodeEnvelope (package:flutter/src/services/message_codecs.dart:651:7)
E/flutter (  696): #1      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:334:18)
E/flutter (  696): <asynchronous suspension>

**Comment by heatherSnepenger** [comment]
@codeTerrain did you figure this out?

**Reply by codeTerrain** [reply]
Yeah @heatherSnepenger, apparently I did. For some strange reasons, turns out the issue had to do with the google services version I was using (4.4.2) in my settings.gradle file. I had to downgrade it to 4.3.15 for it to work. Find attached image. 
<img width="649" alt="Screenshot 2024-11-13 at 7 46 57 AM" src="https://github.com/user-attachments/assets/d0a8a6c6-349a-41ad-bdf3-4d7522668f76">
